### PR TITLE
Fix compilation warning

### PR DESCRIPTION
**Pull request by fwyzard**
Fix clang warning: integer conversion resulted in a change of sign.

### CODE REVIEW

File: DataFormats/TrackReco/interface/TrackExtraBase.h [hunk 1]
```diff
@@ -15,25 +16,29 @@
...
 class TrackExtraBase
 {
-
 public:
     using TrajParams = std::vector<LocalTrajectoryParameters>;
     using Chi2sFive = std::vector<unsigned char>;
 
     /// default constructor
-    TrackExtraBase() : m_firstHit(-1), m_nHits(0) { }
+    TrackExtraBase() : m_firstHit((unsigned int) -1), m_nHits(0) { }
```
Comment (fwyzard) [inline DataFormats/TrackReco/interface/TrackExtraBase.h]: would it be better to initialise this to `0` instead ?

File: DataFormats/TrackReco/interface/TrackExtraBase.h [hunk 2]
```diff
@@ -44,9 +49,6 @@ class TrackExtraBase
         return m_nHits;
     }
 
-    /// accessor to RecHits
-    auto recHits() const { return TrackingRecHitRange(recHitsBegin(), recHitsEnd()); }
```
Comment (perrotta) [inline DataFormats/TrackReco/interface/TrackExtraBase.h]: This was added recently by #25591
What's the reason to remove it?

Comment (fwyzard) [reply DataFormats/TrackReco/interface/TrackExtraBase.h]: I didn't intend to - strange that it is missing, I though started from the master branch as of this morning.

Comment (fwyzard) [reply DataFormats/TrackReco/interface/TrackExtraBase.h]: Added back.

**Comment by fwyzard** [comment]
Note that the widespread whitespace changes are due to fixing the line endings from OSX to Unix.

**Comment by fwyzard** [comment]
@cmsbuild, please test

File: DataFormats/TrackReco/interface/TrackExtraBase.h (hunk 1 above)
Comment (perrotta) [reply DataFormats/TrackReco/interface/TrackExtraBase.h]: It is evident that ```m_firstHit``` is never used at its initialization value: therefore either "0" or the number corresponding to the maximum size result in the same final behavior. Maybe, having 0 hits and the first one located in the maximum possible unsigned int number as initialization values can make sense, even though I agree with you that initializing an unsigned type to "-1" can look odd.
However, your proposed fix silences clang without modifying anything in what was originally there: I would stick on it.

**Comment by perrotta** [comment]
+1
- technical
- jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Pull request merged**